src/frontend/web_application/src/components/MessageList/components/Message/presenter.jsx

```diff
@@ -6,47 +6,24 @@ import ContactAvatarLetter from '../../../ContactAvatarLetter';
 import Dropdown, { withDropdownControl } from '../../../../components/Dropdown';
 import Button from '../../../Button';
 import Icon from '../../../Icon';
+import TextBlock from '../../../TextBlock';
 import MultidimensionalPi from '../../../MultidimensionalPi';
 
 import MessageActionsContainer from '../MessageActionsContainer';
 
 import './style.scss';
 
+const FOLD_HEIGHT = 80; // = .m-message__content--fold height
+
 const DropdownControl = withDropdownControl(Button);
 
-const MessageInfosContainer = ({ __, message, author, locale }) => {
-  const typeTranslations = {
-    email: __('message-list.message.protocol.email'),
-  };
+function generateStateFromProps(props) {
+  const { message } = props;
 
-  return (
-    <div className="m-message__infos-container">
-      <div className="m-message__author">{author.address}</div>
-      {message.type &&
-        (<div className="m-message__type">
-          <span className="m-message__type-label">
-            {__('message-list.message.by', { type: typeTranslations[message.type] })}
-          </span>
-          {' '}
-          <Icon type={message.type} className="m-message__type-icon" spaced />
-        </div>
-      )}
-      <Moment className="m-message__date" format="LT" locale={locale}>
-        {message.date}
-      </Moment>
-    </div>
-  );
-};
-
-MessageInfosContainer.propTypes = {
-  author: PropTypes.shape({}).isRequired,
-  message: PropTypes.shape({}).isRequired,
-  locale: PropTypes.string,
-  __: PropTypes.func.isRequired,
-};
-MessageInfosContainer.defaultProps = {
-  locale: undefined,
-};
+  return {
+    isUnread: message.is_unread,
```

Comment:
is_unread should not be in state: it is always a property of a message.

When an unread message is displayed the action set_read is sent. Then a middleware refresh the message and then you well see message.is_unread property up to date in props.

Finally, state should only be used to temporary set unsaved values (forms) or set a UI state depending on user interactions.